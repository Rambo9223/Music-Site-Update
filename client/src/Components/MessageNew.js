import { Pen } from "react-bootstrap-icons"
/* The link and message body when the admin wants to compose a new email */
export default function MessageNew(props){
    let date = new Date();
    let body = `Dear,
    My name is Scott Ramsay.
    Many Thanks
    Scott`
    return( 
    <>
    <a id="new-email" href={`mailto:?subject=${`Scott Ramsay Enquiry - ${date.toISOString().slice(0,10)}`}&body=${encodeURIComponent(body)}`}>{props.bool?null:<>Compose </>} <Pen/></a>
    </>
    )
}