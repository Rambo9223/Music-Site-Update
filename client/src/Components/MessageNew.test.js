/**
*@jest-environment jsdom
*/
import { render,cleanup} from '@testing-library/react';
import * as React from 'react';
import MessageNew from './MessageNew';

afterEach(()=>{
    cleanup();
});

describe('Message New Test', () => {

    test('Page matches snapshot and element exists', async () => {
        let page = render(<MessageNew/>)
        expect(page).toMatchSnapshot();
        let link = await page.findByText("Compose",{exact:false});
        expect(link).toBeInTheDocument();
    })
})