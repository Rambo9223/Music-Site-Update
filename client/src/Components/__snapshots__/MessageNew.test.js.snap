// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message New Test Page matches snapshot and element exists 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        href="mailto:?subject=Scott Ramsay Enquiry - 2025-01-15&body=Dear%2C%0A%20%20%20%20My%20name%20is%20Scott%20Ramsay.%0A%20%20%20%20Many%20Thanks%0A%20%20%20%20Scott"
        id="new-email"
      >
        Compose 
         
        <svg
          fill="currentColor"
          height="1em"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m13.498.795.149-.149a1.207 1.207 0 1 1 1.707 1.708l-.149.148a1.5 1.5 0 0 1-.059 2.059L4.854 14.854a.5.5 0 0 1-.233.131l-4 1a.5.5 0 0 1-.606-.606l1-4a.5.5 0 0 1 .131-.232l9.642-9.642a.5.5 0 0 0-.642.056L6.854 4.854a.5.5 0 1 1-.708-.708L9.44.854A1.5 1.5 0 0 1 11.5.796a1.5 1.5 0 0 1 1.998-.001m-.644.766a.5.5 0 0 0-.707 0L1.95 11.756l-.764 3.057 3.057-.764L14.44 3.854a.5.5 0 0 0 0-.708z"
          />
        </svg>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      href="mailto:?subject=Scott Ramsay Enquiry - 2025-01-15&body=Dear%2C%0A%20%20%20%20My%20name%20is%20Scott%20Ramsay.%0A%20%20%20%20Many%20Thanks%0A%20%20%20%20Scott"
      id="new-email"
    >
      Compose 
       
      <svg
        fill="currentColor"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m13.498.795.149-.149a1.207 1.207 0 1 1 1.707 1.708l-.149.148a1.5 1.5 0 0 1-.059 2.059L4.854 14.854a.5.5 0 0 1-.233.131l-4 1a.5.5 0 0 1-.606-.606l1-4a.5.5 0 0 1 .131-.232l9.642-9.642a.5.5 0 0 0-.642.056L6.854 4.854a.5.5 0 1 1-.708-.708L9.44.854A1.5 1.5 0 0 1 11.5.796a1.5 1.5 0 0 1 1.998-.001m-.644.766a.5.5 0 0 0-.707 0L1.95 11.756l-.764 3.057 3.057-.764L14.44 3.854a.5.5 0 0 0 0-.708z"
        />
      </svg>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
